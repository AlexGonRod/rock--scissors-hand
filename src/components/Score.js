import React from 'react';

const Score = ({text, className, score}) => {
    return (
        <div className="score-box">
            <span>{text}</span>
            <div className={className}>{score}</div>
        </div>
    )
}

export default Score
