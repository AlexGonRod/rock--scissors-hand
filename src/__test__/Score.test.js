import React from 'react';
import {shallow} from 'enzyme';
import '@testing-library/jest-dom'
import Score from '../components/Score';

describe('<Score />', () => {
    let wrapper = shallow(<Score />);
    beforeEach(() => {
        wrapper = shallow(<Score />);
    });
    test('<Score /> renders', () => {
        expect(wrapper).toMatchSnapshot();
    })
    test('<Score /> text prop', () => {
        const text='Score';

        const wrapper = shallow(<Score text={text}/>)
        expect(wrapper.find('.score-box span').text().trim()).toBe(text);
    })
    test('<Score /> className prop', () => {
        const className='test-class';
        
        const wrapper = shallow(<Score className={className}/>)
        expect(wrapper.find('.score-box .test-class').length).toEqual(1);
    })
    test('<Score /> score prop', () => {
        const score= '0';
        const className='test-class';

        const wrapper = shallow(<Score className={className} score={score}/>)
        expect(wrapper.find('.score-box .test-class').text().trim()).toEqual(score);
    })
});