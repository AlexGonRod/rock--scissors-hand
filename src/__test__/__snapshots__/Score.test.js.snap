// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Score /> <Score /> renders 1`] = `
<div
  className="score-box"
>
  <span />
  <div />
</div>
`;
